test/v2/services/MovementsServiceSpec.scala

```diff
@@ -343,19 +368,44 @@ class MovementsServiceSpec
     val upstreamErrorResponse: Throwable = UpstreamErrorResponse("Internal service error", INTERNAL_SERVER_ERROR)
 
     "on a successful creation, should return a Right" in {
-      when(mockConnector.postMovement(EORINumber(any[String]), any(), eqTo(validRequest))(any[HeaderCarrier], any[ExecutionContext]))
-        .thenReturn(Future.successful(MovementResponse(MovementId("ABC"), MessageId("123"))))
-      val result                                               = sut.createMovement(EORINumber("1"), MovementType.Arrival, validRequest)
-      val expected: Either[PersistenceError, MovementResponse] = Right(MovementResponse(MovementId("ABC"), MessageId("123")))
+      when(mockConnector.postMovement(EORINumber(any[String]), any(), eqTo(Some(validRequest)))(any[HeaderCarrier], any[ExecutionContext]))
+        .thenReturn(Future.successful(MovementResponse(MovementId("ABC"), Some(MessageId("123")))))
+      val result                                               = sut.createMovement(EORINumber("1"), MovementType.Arrival, Some(validRequest))
+      val expected: Either[PersistenceError, MovementResponse] = Right(MovementResponse(MovementId("ABC"), Some(MessageId("123"))))
+      whenReady(result.value) {
+        _ mustBe expected
+      }
+    }
+
+    "on a failed creation, should return a Left with an UnexpectedError" in {
+      when(mockConnector.postMovement(EORINumber(any[String]), any(), eqTo(Some(invalidRequest)))(any[HeaderCarrier], any[ExecutionContext]))
+        .thenReturn(Future.failed(upstreamErrorResponse))
+      val result                                               = sut.createMovement(EORINumber("1"), MovementType.Arrival, Some(invalidRequest))
+      val expected: Either[PersistenceError, MovementResponse] = Left(PersistenceError.UnexpectedError(Some(upstreamErrorResponse)))
+      whenReady(result.value) {
+        _ mustBe expected
+      }
+    }
+  }
+
+  "Create Arrival notification for Large Messages" - {
```

Comment:
we haven't implemented this yet, so how is it working? Is there something wrong with how we're testing for a large message here?

Reply:
We're setting any[HeaderCarrier] but we need to ensure that we aren't sending a content-type header, would setting it to eqTo instead work?

Reply:
For transitmovements, it deals with based on movement - as the response for messageID is optional now - which impact both Departure and Arrival